### PR TITLE
150627361 Blog Posts

### PR DESCRIPTION
**Pull request by emugaya**
### What does this PR do?
Adds Blog to Health checks
### Description of Task to be completed?
Added Blog to healthchecks. Users are now able to create, delete, and update blog posts.

### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/150627374

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,10 @@ hc.sqlite
...
 .DS_Store
-static/.DS_Store
\ No newline at end of file
+static/.DS_Store
+hc/blogs/tests/test_blog_frontend.py
+geckodriver
+geckodriver.log
+hc/blogs/geckodriver
+hc/blogs/tests/chromedriver
+hc/blogs/tests/geckodriver
```
Comment (NgangaMaryanne) [inline .gitignore]: why are these here?

File: hc/blogs/models.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    
+    def publish(self):
+        self.published_date = timezone.now()
+        self.save()
+    
+    def __str__(self):
+        return self.title
+
+    def get_tag_names(self, pk):
+        return Tag.objects.filter(post= pk)
```
Comment (NgangaMaryanne) [inline hc/blogs/models.py]: fix the space around operator

File: hc/blogs/tests/test_blog_post.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		self.assertEquals(response.status_code, 200)
+
+	def test_create_blog_post(self):
+		Post.objects.create(title="Trravelling",text="Nairobi", \
+				tags="Cities", author=self.alice)
+		response = self.client.get('/blog/')
+		self.assertEquals(response.status_code, 200)
+		post = Post.objects.all()
+		self.assertEquals(len(post), 3)
+		response = self.client.get('/blog/post/3/')
```
Comment (NgangaMaryanne) [inline hc/blogs/tests/test_blog_post.py]: what did you intend to do with the response here?

File: hc/blogs/views.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	if request.user == delete_post.author:
+		delete_post.delete()
+		print('Post Deleted')
+		return redirect ('hc-blogs')
+	else:
+		return redirect('hc-blog-detail', pk=delete_post.pk)
+
+	return redirect('hc-blog-detail', pk=delete_post.pk)
+
+def publish_post(request):
```
Comment (NgangaMaryanne) [inline hc/blogs/views.py]: What did you intend this to do?

File: .gitignore (hunk 1 above)
Comment (emugaya) [reply .gitignore]: I was avoiding to commit these in git. I think I need to remove them from the file and stop tracking them on git

File: hc/blogs/models.py (hunk 2 above)
Comment (emugaya) [reply hc/blogs/models.py]: Done.

File: hc/blogs/tests/test_blog_post.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		post = Post.objects.first()
+		post_id = post.id
+		post.title = "Travelling to Kenya"
+		post.text = "Visit Nairobi"
+		post.save()
+		post = Post.objects.get(id=post_id)
+		self.assertEquals(post.title, "Travelling to Kenya")
+		self.assertEquals(post.text, "Visit Nairobi")
+
+
```
Comment (NgangaMaryanne) [inline hc/blogs/tests/test_blog_post.py]: This is too much white space. One blank line is all that is required.

Comment (emugaya) [reply hc/blogs/tests/test_blog_post.py]: White space deleted

File: hc/blogs/views.py (hunk 4 above)
Comment (emugaya) [reply hc/blogs/views.py]: This was supposed to publish the post. I have removed the feature to automatically publish the post once you save.

File: hc/blogs/models.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+from django.db import models
+from django.utils import timezone
+from taggit.models import Tag
+from taggit.managers import TaggableManager
+
+class Post(models.Model):
```
Comment (NgangaMaryanne) [inline hc/blogs/models.py]: This class name is a little confusing. Please use a more descriptive name (something related to blog.)

File: hc/blogs/views.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+			return redirect('hc-blog-detail', pk=post.pk)
+	else:
+		form = PostForm(instance=post)
+	return render(request, 'blog/post_edit.html', {'form':form})
+
+def delete_post(request, pk):
+	delete_post = get_object_or_404(Post, pk=pk)
+	if request.user == delete_post.author:
+		delete_post.delete()
+		print('Post Deleted')
```
Comment (NgangaMaryanne) [inline hc/blogs/views.py]: Print statement is unnecessary. Please remove it.

**Comment by NgangaMaryanne** [comment]
LGTM
